src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -1101,222 +406,211 @@ namespace JSONHelpers
                 }
-                info.cellChildMap().push_back(JsonArrayToNumVec(cellsArray));
             }
 
-            return info;
+            return std::move(customZoneSetsMap);
         }
         catch (const winrt::hresult_error&)
         {
-            return std::nullopt;
+            return {};
         }
     }
 
-    json::JsonObject CustomZoneSetJSON::ToJson(const CustomZoneSetJSON& customZoneSet)
+    json::JsonArray SerializeCustomZoneSets(const TCustomZoneSetsMap& customZoneSetsMap)
     {
-        json::JsonObject result{};
+        json::JsonArray customZoneSetsJSON{};
 
-        result.SetNamedValue(L"uuid", json::value(customZoneSet.uuid));
-        result.SetNamedValue(L"name", json::value(customZoneSet.data.name));
-        switch (customZoneSet.data.type)
-        {
-        case CustomLayoutType::Canvas:
+        for (const auto& [zoneSetId, zoneSetData] : customZoneSetsMap)
         {
-            result.SetNamedValue(L"type", json::value(L"canvas"));
-
-            CanvasLayoutInfo info = std::get<CanvasLayoutInfo>(customZoneSet.data.info);
-            result.SetNamedValue(L"info", CanvasLayoutInfo::ToJson(info));
-
-            break;
+            customZoneSetsJSON.Append(CustomZoneSetJSON::ToJson(CustomZoneSetJSON{ zoneSetId, zoneSetData }));
         }
-        case CustomLayoutType::Grid:
-        {
-            result.SetNamedValue(L"type", json::value(L"grid"));
 
-            GridLayoutInfo gridInfo = std::get<GridLayoutInfo>(customZoneSet.data.info);
-            result.SetNamedValue(L"info", GridLayoutInfo::ToJson(gridInfo));
-
-            break;
-        }
-        }
+        return customZoneSetsJSON;
+    }
 
-        return result;
+    void SerializeDeviceInfoToTmpFile(const JSONHelpers::DeviceInfoJSON& deviceInfo, std::wstring_view tmpFilePath)
+    {
+        json::JsonObject deviceInfoJson = JSONHelpers::DeviceInfoJSON::ToJson(deviceInfo);
+        json::to_file(tmpFilePath, deviceInfoJson);
     }
 
-    std::optional<CustomZoneSetJSON> CustomZoneSetJSON::FromJson(const json::JsonObject& customZoneSet)
+    std::optional<DeviceInfoJSON> ParseDeviceInfoFromTmpFile(std::wstring_view tmpFilePath)
     {
-        try
+        std::optional<DeviceInfoJSON> result{ std::nullopt };
+		if (std::filesystem::exists(tmpFilePath))
         {
-            CustomZoneSetJSON result;
-
-            result.uuid = customZoneSet.GetNamedString(L"uuid");
-            if (!isValidGuid(result.uuid))
+            if (auto zoneSetJson = json::from_file(tmpFilePath); zoneSetJson.has_value())
             {
-                return std::nullopt;
+                if (auto deviceInfo = JSONHelpers::DeviceInfoJSON::FromJson(zoneSetJson.value()); deviceInfo.has_value())
+                {
+                    result = std::move(deviceInfo);
+                }
             }
+        }
+        // TODO(stefan): Check why this was inside triple if above
```

Comment:
So, did you find out? :)

Reply:
No clue :) No regressions so far because of this one. Probably there was a reason initially, which was gone at some point :)

Reply:
Deleted